Build failure: gnucash

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

`environment.systemPackages = (with pkgs; [gnucash];`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/290744650

### Relevant log output

```console
[...]
[  7%] [32m[1mLinking CXX executable ../../../bin/test-import-backend[0m
cd /build/gnucash-5.10/build/gnucash/import-export/test && /nix/store/01dapiw4pjrc9s0166dcyg9xkw0rv01z-cmake-3.31.5/bin/cmake -E cmake_link_script CMakeFiles/test-import-backend.dir/link.txt --verbose=y
/nix/store/ii75mhh7sxl11167m1b86p0qrjsjyjmd-gcc-wrapper-14-20241116/bin/g++ -Werror -Wall -Wmissing-declarations  -O3 -DNDEBUG -Wl,--dependency-file=CMakeFiles/test-import-backend.dir/link.d "CMakeFiles/test-import-backend.dir/gtest-import-backend.cpp.o" "CMakeFiles/test-import-backend.dir/__/import-backend.cpp.o" "CMakeFiles/test-import-backend.dir/__/import-settings.cpp.o" "CMakeFiles/test-import-backend.dir/__/import-utilities.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/mocks/gmock-qofinstance.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/app-utils/mocks/gmock-gnc-prefs.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/mocks/gmock-qofbook.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/mocks/gmock-Account.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/mocks/gmock-Transaction.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/mocks/gmock-Split.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/mocks/fake-qofquery.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/gnc-numeric.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/gnc-rational.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/gnc-int128.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/gnc-date.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/gnc-datetime.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/gnc-timezone.cpp.o" "CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/core-utils/gnc-locale-utils.cpp.o" -o ../../../bin/test-import-backend  /nix/store/gk62b5gxc70dprv92a767zamz5ab27dq-boost-1.87.0/lib/libboost_date_time.so.1.87.0 /nix/store/gk62b5gxc70dprv92a767zamz5ab27dq-boost-1.87.0/lib/libboost_filesystem.so.1.87.0 /nix/store/gk62b5gxc70dprv92a767zamz5ab27dq-boost-1.87.0/lib/libboost_locale.so.1.87.0 /nix/store/gk62b5gxc70dprv92a767zamz5ab27dq-boost-1.87.0/lib/libboost_program_options.so.1.87.0 /nix/store/gk62b5gxc70dprv92a767zamz5ab27dq-boost-1.87.0/lib/libboost_regex.so.1.87.0 -L/nix/store/w1nz8qqc1l0dk6af9s36zchba3hm04ry-glib-2.82.4/lib -Wl,--export-dynamic -lgmodule-2.0 -pthread -lglib-2.0 -L/nix/store/w1nz8qqc1l0dk6af9s36zchba3hm04ry-glib-2.82.4/lib -lgthread-2.0 -pthread -lglib-2.0 -L/nix/store/jbnm36wq89c7iws6xx6xvv75h0drv48x-icu4c-76.1/lib -licui18n -lgmock /nix/store/l2wpczsvddffs7mv083016wq1vzw4ijl-gtest-1.16.0/lib/libgtest.so /nix/store/lfjyqlk2q4dk1idfq6lfdqijflx6khqs-gtk+3-3.24.48/lib/libgtk-3.so /nix/store/lfjyqlk2q4dk1idfq6lfdqijflx6khqs-gtk+3-3.24.48/lib/libgdk-3.so /nix/store/bb9h0qjkl5fh6yr48lw3930gdwvackh5-zlib-1.3.1/lib/libz.so /nix/store/j63vwq8lhj4rgix8an1bqnhlsis3qfma-pango-1.56.1/lib/libpangocairo-1.0.so /nix/store/j63vwq8lhj4rgix8an1bqnhlsis3qfma-pango-1.56.1/lib/libpango-1.0.so /nix/store/33pqwy7v52hkknizv779l8gly58rkb4h-harfbuzz-10.2.0/lib/libharfbuzz.so /nix/store/g3sa64l6qmhm94m7mvccip0mrsphs3yx-at-spi2-core-2.54.1/lib/libatk-1.0.so /nix/store/84hyxhqj8ly2iwlzki5yil6nsz5w4ba3-cairo-1.18.2/lib/libcairo-gobject.so /nix/store/84hyxhqj8ly2iwlzki5yil6nsz5w4ba3-cairo-1.18.2/lib/libcairo.so /nix/store/pwx85z3qpqcg1y1dsgnqv61w2mkk2xnw-gdk-pixbuf-2.42.12/lib/libgdk_pixbuf-2.0.so /nix/store/w1nz8qqc1l0dk6af9s36zchba3hm04ry-glib-2.82.4/lib/libgio-2.0.so /nix/store/w1nz8qqc1l0dk6af9s36zchba3hm04ry-glib-2.82.4/lib/libgobject-2.0.so /nix/store/w1nz8qqc1l0dk6af9s36zchba3hm04ry-glib-2.82.4/lib/libglib-2.0.so /nix/store/gk62b5gxc70dprv92a767zamz5ab27dq-boost-1.87.0/lib/libboost_atomic.so.1.87.0 /nix/store/gk62b5gxc70dprv92a767zamz5ab27dq-boost-1.87.0/lib/libboost_chrono.so.1.87.0 /nix/store/gk62b5gxc70dprv92a767zamz5ab27dq-boost-1.87.0/lib/libboost_system.so.1.87.0 /nix/store/gk62b5gxc70dprv92a767zamz5ab27dq-boost-1.87.0/lib/libboost_thread.so.1.87.0 -lgthread-2.0 -licui18n -lgmock /nix/store/l2wpczsvddffs7mv083016wq1vzw4ijl-gtest-1.16.0/lib/libgtest_main.so
/nix/store/9g4gsby96w4cx1i338kplaap0x37apdf-binutils-2.43.1/bin/ld: CMakeFiles/test-import-backend.dir/__/__/__/libgnucash/engine/gnc-date.cpp.o: undefined reference to symbol 'u_errorName_76'
/nix/store/9g4gsby96w4cx1i338kplaap0x37apdf-binutils-2.43.1/bin/ld: /nix/store/jbnm36wq89c7iws6xx6xvv75h0drv48x-icu4c-76.1/lib/libicuuc.so.76: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[3]: *** [gnucash/import-export/test/CMakeFiles/test-import-backend.dir/build.make:415: bin/test-import-backend] Error 1
make[3]: Leaving directory '/build/gnucash-5.10/build'
make[2]: *** [CMakeFiles/Makefile2:8909: gnucash/import-export/test/CMakeFiles/test-import-backend.dir/all] Error 2
make[2]: Leaving directory '/build/gnucash-5.10/build'
make[1]: *** [CMakeFiles/Makefile2:3703: CMakeFiles/check.dir/rule] Error 2
make[1]: Leaving directory '/build/gnucash-5.10/build'
make: *** [Makefile:199: check] Error 2
```

### Additional context

[vyhgmg55vhcwk7g2qrjiyk7n0ly59z55-gnucash-5.10.log](https://github.com/user-attachments/files/18933162/vyhgmg55vhcwk7g2qrjiyk7n0ly59z55-gnucash-5.10.log)

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.3, NixOS, 25.05 (Warbler), 25.05.20250218.73cf49b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/kdynjy1mbgkdg4p196v9gx6ljpf7q4nk-source`

### Notify maintainers

@domenkozar, @AndersonTorres, @rski, @nevivurn

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/gn/gnucash/package.nix
+++ b/pkgs/by-name/gn/gnucash/package.nix
@@ -3,7 +3,7 @@
   stdenv,
   fetchFromGitHub,
   fetchurl,
-  fetchpatch2,
+  fetchpatch,
   aqbanking,
   boost,
   cmake,
@@ -80,6 +80,12 @@ stdenv.mkDerivation rec {
     ./0003-remove-valgrind.patch
     # this patch makes gnucash exec the Finance::Quote wrapper directly
     ./0004-exec-fq-wrapper.patch
+    # this patch fixes the build against icu 76
+    (fetchpatch {
+      name = "icu-76.patch";
+      url = "https://github.com/Gnucash/gnucash/commit/579eed1facc0f7834ea70b1a342ebca0f125d788.patch";
+      hash = "sha256-XzRPHDPxt2TfeqO48vaNEcfF58neVDMxVqkPKrG5xoM=";
+    })
   ];
 
   # this needs to be an environment variable and not a cmake flag to suppress

